### DISCUSSION TITLE
Bootstrap 5 theme application preparation

### DISCUSSION THREAD

**Discussion by fu-sen** [mkdocs/mkdocs#2336] [General]
Bootstrap is under development for Bootstrap 5, so you need to be careful to apply it in your theme:
https://github.com/twbs/bootstrap/releases

For `mkdocs-bootswatch`, Bootswatch is already preparing a v5 theme:
https://github.com/thomaspark/bootswatch/tree/v5

**Comment by fu-sen** [comment]
You should consider this task.

Bootstrap v5.0.0 has been released.
https://github.com/twbs/bootstrap/releases/tag/v5.0.0

The Bootswatch theme admin was working on it in advance and switched to v5-based within hours of the release of Bootstrap v5.
https://github.com/thomaspark/bootswatch/commit/fe06eb78c9505ca5fc5746fd1ab71abd8ff47642

**Comment by waylan** [comment]
I personally do not have any time to work on this. Of course, contributions are welcome.

**Comment by fu-sen** [comment]
Bootstrap has been updated to v5.0.1.
https://github.com/twbs/bootstrap/releases/tag/v5.0.1

Bootswatch should follow this. Maybe the update will continue for a while.
This will increase our work, so we can wait a little longer.

**Comment by redtide** [comment]
To MKDocs developers: I have some local changes against the main theme targeting for Bootstrap v5.3, which is still in alpha stage but introduced color/theme mode support: would be OK having a MKDocs theme with v5 support to keep for a while in some "testing stage"?

TBH web development is not my forte, would there be anyone interested in this, willing to supervise/help?

If accepted, which name to use for such theme? I would keep the current one based on v4, not replaced with v5.

IMO adding a common theme management also for ReadTheDocs would mean to loose bootstrap 5' theme management features, so I'm not sure if you are OK to handling them differently in this context. Waylan said that both themes should support this feature, but I'm not sure if it's necessary since it's a theme specific feature, not related to MKDocs core. I could try to apply a solution to ReadTheDocs, though personally I'm not interested in that theme. (IMHO both should stay in their own repositories and keep only the main one as git submodule, but that's another story).
